### PR TITLE
Allow user to modify text key

### PR DESCRIPTION
**Pull request by Jaland**
## Issue
[1722](https://github.com/langchain4j/langchain4j/issues/1722)

## Change
Parameterized the key used to look up the text value on weaviate. It should be 

## General checklist

- [x ] There are no breaking changes
- [? ] I have added unit and integration tests for my change (Based on the test that are there I am not really sure what to add that would really test it well)
- [x ] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [x ] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for adding new model integration

- [ ] I have added my new module in the [BOM](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-bom/pom.xml)

## Checklist for adding new embedding store integration

- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added my new module in the [BOM](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-bom/pom.xml)

## Checklist for changing existing embedding store integration

- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

**Comment by Jaland** [comment]
I was also able to test/validate this works with my local code.

File: langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java [hunk 1]
```diff
@@ -80,7 +81,8 @@ public WeaviateEmbeddingStore(
             String objectClass,
             Boolean avoidDups,
             String consistencyLevel,
-            Collection<String> metadataKeys
+            Collection<String> metadataKeys,
+            String textKey
```
Comment (langchain4j) [inline langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java]:
```suggestion
            String textFieldName
```

**Review by ghost — COMMENTED** [review]
@Jaland thanks!

File: langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java (hunk 1 above)
Comment (Jaland) [reply langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java]: Yeah I like that name a lot better than textKey, let me fix that now

**Comment by Jaland** [comment]
Field name updated

**Comment by Jaland** [comment]
I also have a secondary branch I want to create a PR for if this gets accepted for updating the metadata field, but it is a little more involved.

File: langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java [hunk 2]
```diff
@@ -67,6 +67,7 @@ public class WeaviateEmbeddingStore implements EmbeddingStore<TextSegment> {
      * @param useGrpcForInserts Use GRPC instead of HTTP for batch inserts only. <b>You still need HTTP configured for search</b>
      * @param securedGrpc       The GRPC connection is secured
      * @param grpcPort          The port, e.g. 50051. This parameter is optional.
+     * @param textFieldName     The key in metadata that contains the text. Default is "text".
```
Comment (langchain4j) [inline langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java]:
```suggestion
     * @param textFieldName     The name of the field that contains the text of a {@link TextSegment}. Default is "text".
```

**Review by ghost — DISMISSED** [review]
@Jaland thank you!

**Pull request merged**